state-recover/appcore/domain-models/src/main/kotlin/build/linea/staterecover/TransactionFromL1RecoveredData.kt

```diff
@@ -0,0 +1,103 @@
+package build.linea.staterecover
+
+import net.consensys.encodeHex
+import java.math.BigInteger
+
+data class TransactionFromL1RecoveredData(
```

Comment:
Can we get away of our own transaction type? Maybe if we parse RLP Besu's Transaction directly, when transaction object is extended (with Prague fork, for example), it would be just a matter of updating Besu dependency on State recovery side. What benefit are we getting from defining our own type?

Reply:
Maybe, but rather keep it like this because:
- this expresses better exactly what can be recovered from L1 and what not. 
- core logic is not coupled to Besu domain model

this way, `importBlocks` method has a very clear boundary if shall be provided. Using besu domain model we would  need to send Fake signatures and so on and this would make behaviour harder to understand.